openeo/extra/job_management.py

```diff
@@ -437,15 +441,38 @@ def _update_statuses(self, df: pd.DataFrame):
                 con = self._get_connection(backend_name)
                 the_job = con.job(job_id)
                 job_metadata = the_job.describe()
-                _log.info(
-                    f"Status of job {job_id!r} (on backend {backend_name}) is {job_metadata['status']!r}"
-                )
+                _log.info(f"Status of job {job_id!r} (on backend {backend_name}) is {job_metadata['status']!r}")
+
+                if job_metadata["status"] == "running":
+
+                    timezone = datetime.timezone.utc
+                    job_time_created = datetime.datetime.strptime(the_job.describe()["created"], "%Y-%m-%dT%H:%M:%SZ")
+                    job_time_created = job_time_created.replace(tzinfo=timezone)
+
+                    if datetime.datetime.now(timezone) > job_time_created + self.max_running_duration:
+                        try:
+                            _log.info(
+                                f"Cancelling job {job_id} on backend {backend_name} as it has been running for more than {self.max_running_duration} minutes"
+                            )
+                            the_job.stop_job()
+                            self.on_job_error(the_job, df.loc[i])
```

Comment:
would this then be a replication of on_job_error except that you might want to log the info logs instead of the error logs? (I am adding the cancel logging to info)